common/data/dataHelpers.ts

```diff
@@ -1,16 +1,73 @@
 export const STRAPI_BASE_URL = 'https://map-api.ekaterinburg.io/api';
 
+const PAGINATION_SIZE = 800;
+const MAX_REQUESTS = 10;
+
 export async function fetchAPI(url: string) {
     const response = await fetch(url);
     return response.json();
 }
 
-export async function getObjectsTotalCount(urlWithoutParams: string) {
-    return (await fetchAPI(`${urlWithoutParams}?pagination[pageSize]=1`))
-        .meta.pagination.total;
+export async function getObjectsTotalCount(url: string) {
+    const hasParams = url.includes('?');
+    return (await fetchAPI(`${url}${hasParams ? '&' : '?'}pagination[pageSize]=1`)).meta.pagination
+        .total;
 }
 
 export function parseJsonWithSingleQuotes(json: string) {
-    if (!json) return json;
+    if (!json) {
+        return json;
+    }
+
     return JSON.parse(json.replaceAll("'", '"'));
 }
+
+export async function parallelRequests<T, R>(url: string, dataMapper: (x: T) => R) {
+    const totalCount = await getObjectsTotalCount(url);
+
+    const pagesCount = Math.ceil(totalCount / PAGINATION_SIZE);
+
+    const requests: Promise<{ data: T[] }>[] = [];
+    for (let i = 1; i <= pagesCount; i++) {
+        requests.push(
+            fetchAPI(`${url}&pagination[pageSize]=${PAGINATION_SIZE}&pagination[page]=${i}`),
```

Comment:
Let's use `new URL` and `searchParams`

```suggestion
        const requestUrl = new URL(url);
        const params = requestUrl.searchParams;
        params.append('pagination[pageSize]', PAGINATION_SIZE);
        params.append('pagination[page]', i);
        requests.push(
            fetchAPI(requestUrl),
```

Reply:
or

```js
for (let param of [
    ['pagination[pageSize]', PAGINATION_SIZE],
    ['pagination[page]', i]
]) {
    requestUrl.searchParams.append(param);
}
```

Reply:
like more readable
but it's a cycle within a cycle)

```suggestion
        const requestUrl = new URL(url);
        const params = {
                'pagination[pageSize]': PAGINATION_SIZE,
                'pagination[page]': i
        };
        for (let key in params) {
                requestUrl.searchParams.append(key, params[key]);
        }
        requests.push(fetchAPI(requestUrl));
```